src/runtime/registry.rs

```diff
@@ -133,3 +160,28 @@ impl Deref for Scope {
         &self.inner
     }
 }
+
+#[derive(Debug, Default)]
+pub struct AddressRegistry {
```

Comment:
Suggestion (ignore if you disagree): maybe `AddressRegistry` is too small of a type to name the module after it. My preference would be to either name this module `scope` and the other which is currently called "scope" `scope_view`, or even merge them into one `scope` module.

Reply:
It's not actually named after the address registry, but I agree it's kind of awkward. I want there to be separation between the public facing scope and the "registry" which is really just a linked tree of scopes. Maybe we could use `scope` instead of `registry` and `runtime_scope` instead of `scope`.

Reply:
yea, that works for me :+1: